src/annotation_processor.md

```diff
@@ -0,0 +1,149 @@
+
+### A common application of overriding annotation using PathSpec
+In the above `EnterpriseUserRecord` example, we use `@persistencePolicyInDays = {"/recycledChatHistories" : 3650}` do denote the field we are overriding is `recycledChatHistories`. This is basically using PathSpec to reference the field or child schema that needs to be overriden. PathSpec <insert-linke-here> can be used as a path to associate relations between fields among nested schemas. Users can use PathSpec to inambigously specifcy all the paths to the child fields that they want to override.
+
+
+In Linkedin, such usage is so common that we want to use 
+we provided `PathSpecBaseVisitor`. 
+
+It assumes \\
+(1) All *overrides* to the fields are specified using PathSpecs to the fields. \\
+(2) All *overrides* are applied on fields, not on the record
+
+For more examples regarding the syntax, one can read the java doc from `PathSpecBaseVisitor`.//<insert Link about > and the test cases //<insert Link about >
+
+
+One can seek to extend this case to adapt to their own use cases. Please see next section regarding what class to extend to fit the best use cases.
+
+## Use the schema annotation processor
+We have created a paradigm of processing Rest.Li schema annotations and wrapped them into `com.linkedin.data.schema.annotation` package.
+First thing to understand is that `DataSchema` object internally is stored recursively. We added `resolvedProperties` attribute, in order to store the final reesolved annotation.
+
+For an example schema:
+
+```pdl
+record Employee {
+  id: long
+  supervisor: Employee
+}
+```
+
+Its memory presentation is as followed:
+<center>
+<b>DataSchema Example</b><br><img src="{{ 'assets/images/DataSchema_with_resolvedProperties.png' | relative_url }}" />
+</center>
+
+The SchemaAnnotationProcessor will process `DataSchema` using `DataSchemaRichContextTraverser` and resolve the annotation for fields in the schemas.
+
+
+<center>
+<b>Sequence Diagram</b><br><img src="{{ 'assets/images/AnnotationProcessor_UML_sequence_diagram.png' | relative_url }}"  />
+</center>
+
+The `DataSchemaRichContextTraverser` travsers the data schema and in turn calls an implementation of `SchemaVisitor`. It is the `SchemaVisitor`' that resolves the annotations for fields in the schemas, based on the context provided by the `DataSchemaRichContextTraverser`. `SchemaVisitor` may also create copy of original data schema if the oroginal data schema needs to be immutable.
```

Comment:
extra quote for SchemaVisitor

oroginal -> original

Comment about immutable not very clear. Can you add more details?